torchdata/dataloader2/dataloader2.py

```diff
@@ -49,7 +51,10 @@ def __next__(self) -> T_co:
         if self.iterator_id == self.dataloader.valid_iterator_id:
             self.dataloader._reset_iter = True
             try:
-                return next(self.dataloader._datapipe_iter)  # type: ignore[arg-type]
+                if self.dataloader._paused:
+                    raise StopIteration
+                else:
+                    return next(self.dataloader._datapipe_iter)  # type: ignore[arg-type]
             except PauseIteration:
```

Comment:
Feeling we can consolidate dataloader._pause and PauseIteration here. DL Iter PauseIteration, RS pause, DL pause should be consistent